app.py

```diff
@@ -32,21 +32,79 @@
     'upstream/Rover-4.2' : 'Rover 4.2 stable'
 }
 default_branch = 'upstream/master'
-CURR_BRANCH = default_branch
 
-def get_boards_from_ardupilot_tree(branch):
+def get_vehicles():
+    return VEHICLES
+
+def get_default_vehicle():
+    return default_vehicle
+
+def get_branches():
+    return BRANCHES
+
+def get_default_branch():
+    return default_branch
+
+# LOCKS
+queue_lock = Lock()
+head_lock = Lock()  # lock git HEAD, i.e., no branch change until this lock is released
+
+def is_valid_branch(branch):
+    return get_branches().get(branch) is not None
+
+def get_git_hash(branch):
+    return subprocess.check_output(['git', 'rev-parse', branch], cwd=sourcedir, encoding='utf-8').rstrip()
```

Comment:
Let's be explicit and pass in `shell=False` here (and on other `subprocess.` calls.  We're passing user-data (yes, vetted user data, but...)

Reply:
I'll do it. Thanks :)